### ISSUE TITLE
selectpicker('mobile') not working on iOS

### ISSUE THREAD

**Issue by sam-thecoder** [snapappointments/bootstrap-select#1828]
I have an issue using the select picker on mobile, I've been able to catch this issue using chrome's mobile device emulator: https://youtu.be/vNzfvwzgPSM it will show it works okay on desktop view but on mobile view you click it multiple times with it not responding accurately.

**Comment by sinneren** [comment]
Confirm it on iOS 11+ and Chrome Dev Tools with Mobile Touch. But on Android and old iOS work well.

**Comment by chrisw73** [comment]
Confirm on iPad Air, iPad Pro and iPhone 7 all with iOS 11.3 and 11.4 and Safari. A native Fallback would be nice...
[Update] Seemt to occur with BS 4 only...

**Comment by sanfeijoo** [comment]
I'm using bootstrap 3.3.7 and it doesn't work in mobile when mobile option is set to true

**Comment by logicsdrive** [comment]
Any solution so far regarding this issue ?

**Comment by caseyjhol** [comment]
No solution yet. I still haven't been able to recreate this issue on my end.

**Comment by logicsdrive** [comment]
@caseyjhol here is the demo of that issue, hoping that will help
http://logicsdrive.us/clients/kuwaitnet/oxygen/selectpicker.html
You can check this link on mobile to see the issue.

**Comment by kveldscholten** [comment]
on my ipad with ios 11.4.1 is the same problem

**Comment by Serzhs** [comment]
My solution is to downgrade bootstrap select version to v1.12.4

**Comment by logicsdrive** [comment]
@Espumisan v1.12.4 works fine with bootstrap 4 ?

**Comment by caseyjhol** [comment]
Still checking into this (need to test on iOS). Works on Android 8.0.0 with both Chrome and Firefox.

**Comment by logicsdrive** [comment]
Yes there is no issue on Android .. Issue is with IOS

**Comment by soberdor** [comment]
I can confirm this problem on iOS 11.4.2 and Bootstrap 4.1

**Comment by jint0136** [comment]
Yeaa I will try

**Comment by caseyjhol** [comment]
This problem was introduced in commit 7e331136b399b6d9f02452abe6aeaab83dd84aa6. The native select menu is opened, but then the button is instantly focused, thus closing the menu. On all other devices, operating systems, and browsers, this works fine (as it happens simultaneously), except in Safari on iOS.

**Comment by caseyjhol** [comment]
Working on a fix, but this is made more difficult by jquery/jquery#1750 and jquery/jquery#1741 (i.e. it's not possible to namespace focus events in jQuery v1.9+).

**Comment by soberdor** [comment]
Hi and thank you for the fix - however, I tested the latest commit and it didn't seem to work. Debugging into the newly introduced "if (!that.options.mobile)" I noticed it was false despite being properly initialized as 'mobile'.

**Comment by caseyjhol** [comment]
@soberdor If you use `$.fn.selectpicker.Constructor.DEFAULTS.mobile = true;` instead of `.selectpicker('mobile')` does it work?

**Comment by soberdor** [comment]
Thanks, yes that works! But is there still any way to use live search on mobile devices? data-live-search="true" doesn't seem to have any effect now.

**Comment by caseyjhol** [comment]
@soberdor The "mobile" option (which should probably be renamed) simply uses the device's native select menu, so until something like live search is supported on devices, it won't be possible using that feature. However, I do have plans to implement a more mobile-friendly version of the menu in a future update.

**Comment by caseyjhol** [comment]
Released in [v1.13.3](https://github.com/snapappointments/bootstrap-select/releases/tag/v1.13.3)!

**Comment by fatheaddrummer** [comment]
Hi and thanks a lot for everything as well! I just wanted to inform you that I still couldn't get the mobile version running on iPhone 5 SE. I used Bootstrap 4.0.0 and Bootstrap-select 1.13.5. I could work around it however with` $.fn.selectpicker.Constructor.DEFAULTS.mobile = true;`

Thanks a lot!

**Comment by kveldscholten** [comment]
I have test Bootstrap-select 1.13.5 on iPhone 5 and iPhone 7 with Bootstrap 3.3.7 but couldn't work for me.

My Code to check Mobile
```
if (/Android|webOS|iPhone|iPad|iPod|BlackBerry/i.test(navigator.userAgent)) {
    $('#salutation').selectpicker('mobile');
} else {
    $('#salutation').selectpicker();
}
```

thanks for help

**Comment by toix** [comment]
Safari 5.1.7 for Windows:
![image](https://user-images.githubusercontent.com/6238218/60093445-b5cc9080-9749-11e9-9e79-fd96a6dfea7c.png)

I know that Safari development for Windows has been abandoned more than 5 years ago. But it's not a good indicator bootstrap-select is the one breaking all JavaScript.

**Comment by torrobinson** [comment]
I'm still getting this for Safari on iOS, but not on anything Android/Desktop. Has there ever been a fix for this? The `$.fn.selectpicker.Constructor.DEFAULTS.mobile = true;` workaround has not worked for me yet, unfortunately.

**Comment by kfe908** [comment]
Does not work when using Android and Firefox, even downgrading does not work :( The problem exists for years and no solution...that's bad!

**Comment by qxygene** [comment]
any fix on this yet?

**Comment by caseyjhol** [comment]
This is working for me with Firefox 68.7.0 on Android, as well as Chrome 81 on Android. Can anybody confirm which browsers/operating systems this happens with? Is it only iOS?

**Comment by qxygene** [comment]
Not working on any safari browser. Tested with browserstack !

**Comment by caseyjhol** [comment]
Was able to get my hands on an iOS device and was able to recreate this issue (native select list opens up and immediately closes).

**Comment by qxygene** [comment]
Best way is for now is to detect mobile and destroy bootstap-seelct i guess !

**Comment by chrisw73** [comment]
> Best way is for now is to detect mobile and destroy bootstap-seelct i guess !

This is, what I'd preffer, because mobile devices usually handle this their way best...

**Comment by caseyjhol** [comment]
Released in [v1.13.17](https://github.com/snapappointments/bootstrap-select/releases/tag/v1.13.17)!
